### PR TITLE
A bit more friendly language. Update based on comments.

### PR DESCRIPTION
**Pull request by jmlundberg**
This is an update of https://github.com/isocpp/CppCoreGuidelines/pull/564 based on the comments there. The only difference with respect to that is that I removed all questioned changes.

Several small edits, not trying to change the meaning.

Trying to make the text more inclusive. For example, few can write a completely correct make_unique. "supercalifragilisticexpialidocious" is funny but distracting.

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -7777,14 +7777,14 @@ This makes the function's ??? explicit.
 
 ##### Reason
 
-Violating this rule is the number one cause of losing reference counts and finding yourself with a dangling pointer.
-Functions should prefer to pass raw pointers and references down call chains.
-At the top of the call tree where you obtain the raw pointer or reference from a smart pointer that keeps the object alive.
-You need to be sure that the smart pointer cannot inadvertently be reset or reassigned from within the call tree below.
+The problem arise if a reference or raw pointer is obtained from an aliased smart pointer which is then inadvertently reset or reassigned from down a call chain.
+
+When it comes to smart pointers, violations of this rule are the number one cause of lost reference counts and dangling pointers.
```
Comment (gdr-at-ms) [inline CppCoreGuidelines.md]: The old text is fine.

File: CppCoreGuidelines.md [hunk 2]
```diff
@@ -8259,13 +8258,19 @@ In each case, we save writing a longish, hard-to-remember type that the compiler
...
 
 ##### Example
 
     auto lst = { 1, 2, 3 };   // lst is an initializer list
     auto x{1};   // x is an int (after correction of the C++14 standard; initializer_list in C++11)
 
+##### Exception
+
+Be very careful with `auto` in combination with certain expression templates and proxy types (Two examples are the [Eigen](http://eigen.tuxfamily.org) math library and [Boost Spirit](http://www.boost.org/doc/libs/1_60_0/libs/spirit/doc/html/index.html)).
+
```
Comment (gdr-at-ms) [inline CppCoreGuidelines.md]: This note is not necessary.

**Comment by jmlundberg** [comment]
Hi GDR. This PR is just exactly what was left after your last round of comments. I prepared this PR based on that, deleting what you rejected then.
I feel strongly about these two parts. 
The first part is quite unclear and the "reasons" were sort of just more rules rather than an explanation. It was clear only if it was already understood. At least that's how it reads to me. But I'm sure there's better ways to explain it.
When it comes to being careful with auto _in some cases_: It's item 6 of Modern Effective C++, where 6 pages already at page 45 are spent on it just after auto is introduced and praised. Both for good reason. Spending this one sentence in these guidelines seem very appropriate.

**Comment by gdr-at-ms** [comment]
Sorry my previous feedback was incomplete  and that I was unavailable right a week after that.

I see two ways to proceed: (1) send an updated PR with the latest feedback and open an issue for clarification for `auto`; or (2) we turn this PR into an issue and I volunteer to take it.

**Comment by tapika** [comment]
I think reference counting is in overall bit old approach - 
http://stackoverflow.com/questions/20896618/a-simple-implementation-of-smart-pointer-class/39154667#39154667

I this answer there is a link with article about smart pointers - I think it makes quite clear statement not to use reference counting.

But I still did not get shared_ptr and unique_ptr - why they are needed ? Simple written class seems to be smaller and simpler than new C++1xx ?!

**Comment by AndrewPardoe** [comment]
Thank you for the suggestion and discussion. Per our editor's meeting this week, we feel that there wasn't enough impetus to make these changes.

**Pull request closed without merging**